src/debug/netcoredbg/variables.cpp

```diff
@@ -481,8 +482,15 @@ HRESULT Variables::Evaluate(
     ToRelease<ICorDebugFrame> pFrame;
     IfFailRet(GetFrameAt(pThread, stackFrame.GetLevel(), &pFrame));
     ToRelease<ICorDebugILFrame> pILFrame;
-    if (pFrame)
-        IfFailRet(pFrame->QueryInterface(IID_ICorDebugILFrame, (LPVOID*) &pILFrame));
+
+    if (!pFrame) {
+        stackFrame = StackFrame(threadId, 0, "");
+        frameId = stackFrame.id;
```

Comment:
This issue should be investigated. In fact, you can't just move to another thread and another frameId just because you want, this is wrong solution. We should find out why we have wrong frameId at this point.
For example, user could provide wrong frameId for "print" command (CLI protocol), and we should care about (provide error), but not evaluate for frameId we want instead.
Note, we switch to `GetLastStoppedThreadId()` only in case user/IDE don't provide frameId at all (see protocols related code).

Reply:
That makes sense. @puremourning You wrote the vimspector code that does this, what would you expect the adapter to do?

Reply:
if the expression is requested in a frame which doesn't exist, return an error